src/components/Combobox/components/ComboboxItems/ComboboxItems.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useMemo } from "react";
+import cx from "classnames";
+import { getOptionsByCategories } from "components/Combobox/ComboboxService";
+import {
+  COMBOBOX_CATEGORY_ITEM,
+  COMBOBOX_DIVIDER_ITEM,
+  COMBOBOX_OPTION_ITEM
+} from "components/Combobox/components/ComboboxConstants";
+import { comboboxItemRenderer } from "components/Combobox/components/ComboboxRenderers";
+import { VirtualizedList } from "components";
+import styles from "./ComboboxItems.modules.scss";
+
+export const ComboboxItems = ({
+  categories,
+  options,
+  filterValue,
+  withCategoriesDivider,
+  optionRenderer,
+  activeItemIndex,
+  isActiveByKeyboard,
+  onOptionClick,
+  onOptionEnter,
+  onOptionLeave,
+  optionLineHeight,
+  shouldScrollToSelectedItem,
+  renderOnlyVisibleOptions
+}) => {
+  let items = useMemo(() => {
+    let items = [];
+    if (categories) {
+      const optionsByCategories = getOptionsByCategories(options, categories, filterValue);
+      let optionIndex = 0;
+      Object.keys(optionsByCategories).forEach((categoryId, categoryIndex) => {
+        const withDivider = withCategoriesDivider && categoryIndex !== 0;
+        if (withDivider) items.push({ type: COMBOBOX_DIVIDER_ITEM, height: 17, id: `${categoryId}-divider` });
```

Comment:
lets extract method like
createDividerItem
createCategoryItem
createOptionItem

Reply:
Also extract heights to constants